src/mscorlib/src/System/Text/ASCIIEncoding.cs

```diff
@@ -78,12 +78,22 @@ public override unsafe int GetByteCount(char* chars, int count)
             return EncodingForwarder.GetByteCount(this, chars, count);
         }
 
-        public override int GetBytes(String chars, int charIndex, int charCount,
-                                              byte[] bytes, int byteIndex)
-        {
-            return EncodingForwarder.GetBytes(this, chars, charIndex, charCount, bytes, byteIndex);
+#if !BIGENDIAN
```

Comment:
CoreLib is build architecture specific for number of different reasons. If you would like the code to be more pretty (and create more work for JIT to get rid of unreachable code), you can do the check dynamically based on a static property and have the property return true/false constant.